contrib/hadoop-store-builder/src/java/voldemort/store/readonly/fetcher/BasicFetchStrategy.java

```diff
@@ -208,6 +216,26 @@ public BasicFetchStrategy(HdfsFetcher fetcher,
                 }
             }
         }
+
+        //second time checksum validation. Check if the local file is consistent with the buffer
+        if (bufferCheckSumGenerator != null){
+            CheckSum fileCheckSum = CheckSum.getInstance(checkSumType);
+            BufferedInputStream in = new BufferedInputStream(new FileInputStream(dest));
+            int read;
+
+            try {
+                while ((read = in.read(buffer)) >= 0) {
+                    fileCheckSum.update(buffer, 0, read);
+                }
+
+                if (ByteUtils.compare(checkSum, fileCheckSum.getCheckSum()) != 0)
+                    throw new VoldemortException("Local file: " + dest.getAbsolutePath() +
```

Comment:
Can you also print what is the fileCheckSum and buffer checksum in the error logs ?